vyper/builtins/functions.py

```diff
@@ -294,6 +294,46 @@ class Slice(BuiltinFunction):
     ]
     _return_type = None
 
+    def evaluate(self, node):
+        validate_call_args(node, 3)
+        literal_value, start, length = node.args
+        if (
+            isinstance(literal_value, (vy_ast.Bytes, vy_ast.Str, vy_ast.Hex))
+            and isinstance(start, vy_ast.Int)
+            and isinstance(length, vy_ast.Int)
+        ):
+            (start_val, length_val) = (start.value, length.value)
+
+            if start_val < 0:
+                raise ArgumentException("Start cannot be negative", start)
+            elif length_val <= 0:
+                raise ArgumentException("Length cannot be negative", length)
+
+            if isinstance(literal_value, vy_ast.Hex):
```

Comment:
i think we should calculate a helper variable which is something like `bytes_value = bytes.fromhex(bytestring.value.removeprefix("0x"))` and then the later computations will get simpler

Reply:
indeed, thanks!